sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -99,17 +99,14 @@ trait DataSourceScanExec extends LeafExecNode {
 
 /** Physical plan node for scanning data from a relation. */
 case class RowDataSourceScanExec(
-    fullOutput: Seq[Attribute],
```

Comment:
can you find out the PR that added it? I can't quite remember why we have it.

Reply:
It was introduced in https://github.com/apache/spark/pull/18600 for plan equality comparison. 
I manually print out the two canonicalized plans for `df1` and `df2` in  https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/RowDataSourceStrategySuite.scala#L68 to check my change.
Before my change: 

*(2) HashAggregate(keys=[none#0], functions=[min(none#0)], output=[none#0, #0])
+- Exchange hashpartitioning(none#0, 5), true, [id=#25]
   +- *(1) HashAggregate(keys=[none#0], functions=[partial_min(none#1)], output=[none#0, none#4])
      +- *(1) Scan JDBCRelation(TEST.INTTYPES) [numPartitions=1] _**[none#0,none#1]**_ PushedFilters: [], ReadSchema: struct<none:int,none:int>

*(2) HashAggregate(keys=[none#0], functions=[min(none#0)], output=[none#0, #0])
+- Exchange hashpartitioning(none#0, 5), true, [id=#52]
   +- *(1) HashAggregate(keys=[none#0], functions=[partial_min(none#1)], output=[none#0, none#4])
      +- *(1) Scan JDBCRelation(TEST.INTTYPES) [numPartitions=1] _**[none#0,none#2]**_ PushedFilters: [], ReadSchema: struct<none:int,none:int>

After my change :

*(2) HashAggregate(keys=[none#0], functions=[min(none#0)], output=[none#0, #0])
+- Exchange hashpartitioning(none#0, 5), true, [id=#25]
   +- *(1) HashAggregate(keys=[none#0], functions=[partial_min(none#1)], output=[none#0, none#4])
      +- *(1) Scan JDBCRelation(TEST.INTTYPES) [numPartitions=1] _**[A#0,B#1]**_ PushedFilters: [], ReadSchema: struct<A:int,B:int>

*(2) HashAggregate(keys=[none#0], functions=[min(none#0)], output=[none#0, #0])
+- Exchange hashpartitioning(none#0, 5), true, [id=#52]
   +- *(1) HashAggregate(keys=[none#0], functions=[partial_min(none#1)], output=[none#0, none#4])
      +- *(1) Scan JDBCRelation(TEST.INTTYPES) [numPartitions=1] _**[A#0,C#2]**_ PushedFilters: [], ReadSchema: struct<A:int,C:int>